cocos/renderer/CCTexture2D.cpp

```diff
@@ -1444,4 +1464,9 @@ void Texture2D::removeSpriteFrameCapInset(SpriteFrame* spriteFrame)
     }
 }
 
+void Texture2D::setAutoGenerateMipmap(bool generateMipmaps)
+{
+    _generateMipmaps = generateMipmaps;
```

Comment:
but you initialize `_hasMipmaps` like this 
```c++
_hasMipmaps = mipmapsNum > 1 || _generateMipmaps;
```
which means `_generateMipmaps` should affect the value of `_hasMipmaps`. That's why i think `setAutoGenerateMipmap ` should change `_hasMipmaps` too, because it may be invoked after `initWithMipmaps `.